app/services/builders/chartkick/department_distribution_data.rb

```diff
@@ -10,21 +10,52 @@ def call
       private
 
       def retrieve_records
-        return review_turnarounds if @query[:name].equal?(:review_turnaround)
+        metric
+          .records_with_departments
+          .where(departments: { id: @entity_id })
+          .where(created_at: @query[:value_timestamp])
+      end
+
+      def metric_name
+        @query[:name]
+      end
+
+      def metric
+        @metric ||= self.class.const_get(metric_name.to_s.camelize).new
+      end
 
-        merge_times
+      def resolve_interval(entity)
+        metric.resolve_interval(entity)
       end
 
-      def review_turnarounds
-        ::CompletedReviewTurnaround.joins(review_request: { project: { language: :department } })
-                                   .where(departments: { id: @entity_id })
-                                   .where(created_at: @query[:value_timestamp])
+      class MergeTime
```

Comment:
why do you prefer to have 3 classes in a single file instead of 1 class per file?

Reply:
https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#project-structure
https://github.com/fxn/zeitwerk#file-structure

Reply:
I did it just for readability, but you're right, I should move them to their own files.